app/src/server/utils/prepareDatasetEntriesForImport.ts

```diff
@@ -29,7 +32,10 @@ export const prepareDatasetEntriesForImport = async (
   existingTrainingCount += entriesToImport.filter((row) => row.split === "TRAIN").length;
 
   const newTotalEntries = existingCount + entriesToImport.length;
-  const numTrainingToAdd = Math.floor(trainingRatio * newTotalEntries) - existingTrainingCount;
+  const numTrainingToAdd = Math.min(
```

Comment:
Curious what drove this change -- I don't immediately see the bug in the code as it was written previously but maybe I'm missing something?

Reply:
It fixed a bug in the previous version of the code, but I didn't see how much you had changed. I think the changes you made fixed the bug that this is supposed to fix, so I can remove it now.